### PR TITLE
🐛 BigQuery source: Fix nested structs

### LINKED ISSUE #5460: Bigquery source connector can't fetch data with nested structs
## Enviroment
- **Airbyte version**: 0.29.8
- **OS Version / Instance**: Windows 10
- **Deployment**: Docker desktop
- **Source Connector and version**: airbyte/source-bigquery 0.1.1
- **Destination Connector and version**: airbyte/destination-snowflake 0.3.12
- **Severity**: Critical
- **Step where error happened**: Sync job

## Current Behavior
Sync stops for BigQuery tables with array of nested structs. Error message:

`Exception in thread "main" java.lang.UnsupportedOperationException: Retrieving field value by name is not supported when there is no fields schema provided`

Array of just struct works fine.

## Expected Behavior
Sync should complete.

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

```

replace this with
your long log
output here

```

</details>

## Steps to Reproduce
1. Create bigquery table with nested struct, such as e.g.
```
with
  bertil as (
    select
      "Bertil" AS name,
      190 as height,
      [struct<key string, value STRUCT<string_val string>>('work_place', struct('N/A')),
       struct<key string, value STRUCT<string_val string>>('country', struct('Sweden'))] as persons),
  linda as (
    select
      "Linda" AS name,
      176 as height,
      [struct<key string, value STRUCT<string_val string>>('work_place', struct('Home')),
       struct<key string, value STRUCT<string_val string>>('country', struct('Sweden'))] as persons),
  combined as (
    select *
    from bertil 
    union all 
    select *
    from linda )
select *
from combined
```
2. Start sync job

## Are you willing to submit a PR?

Not at this point.

### PR DESCRIPTION
**Pull request by irynakruk**
## What
Sync stops for BigQuery tables with array of nested structs. Error message:
Exception in thread "main" java.lang.UnsupportedOperationException: Retrieving field value by name is not supported when there is no fields schema provided
Array of just struct works fine.

## How
Modified `Record` parse method

## Recommended reading order
1. `BigQuerySourceOperations.java`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [x] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

### CODE REVIEW

**Review by etsybaev — APPROVED** [review]

**Review by alexandr-shegeda — APPROVED** [review]

File: airbyte-integrations/connectors/source-bigquery/Dockerfile [hunk 1]
```diff
@@ -9,5 +9,5 @@ COPY build/distributions/${APPLICATION}*.tar ${APPLICATION}.tar
 RUN tar xf ${APPLICATION}.tar --strip-components=1
 
 # Airbyte's build system uses these labels to know what to name and tag the docker images produced by this Dockerfile.
-LABEL io.airbyte.version=0.1.1
+LABEL io.airbyte.version=0.1.2
 LABEL io.airbyte.name=airbyte/source-bigquery
```
Comment (davinchia) [inline airbyte-integrations/connectors/source-bigquery/Dockerfile:13]: nit: I know this wasn't checked in by you, but can you add a new line here. thanks

**Review by davinchia — APPROVED** [review]
Nice!

Before we publish and merge, can you run the test command to make sure this passes all tests?

In the future, can you also make sure to go through the updating connector checklist before I review? It's make it easier for me to know what has been done/what to focus on. Thanks!

File: airbyte-integrations/connectors/source-bigquery/Dockerfile (hunk 1 above)
Comment (irynakruk) [reply airbyte-integrations/connectors/source-bigquery/Dockerfile:13]: Sure, done

**Comment by irynakruk** [comment]
> Nice!
> 
> Before we publish and merge, can you run the test command to make sure this passes all tests?
> 
> In the future, can you also make sure to go through the updating connector checklist before I review? It's make it easier for me to know what has been done/what to focus on. Thanks!

Sure, thanks, I'll update connector's checklist, so all will be visible there. I've run test command, also will mark it in checklist.

**Pull request merged**